lms/djangoapps/course_api/api.py

```diff
@@ -48,7 +47,7 @@ def course_detail(request, username, course_key):
         course = get_course_with_access(user, 'see_exists', course_key)
     except Http404:
         raise NotFound()
-    return CourseSerializer(course, context={'request': request}).data
+    return course
```

Comment:
since return types for this function (and `list_courses`) have changed, please update docstrings correspondingly.